src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -211,4 +216,22 @@ public ResponseEntity<Void> resetRepository(@PathVariable Long participationId,
 
         return ResponseEntity.ok().build();
     }
+
+    @GetMapping("programming-exercise-participations/{participationId}/commits-info")
+    @EnforceAtLeastInstructor
+    List<CommitInfoDTO> getCommitInfosForParticipationRepo(@PathVariable long participationId) {
+        ProgrammingExerciseStudentParticipation participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
+        participationAuthCheckService.checkCanAccessParticipationElseThrow(participation);
```

Comment:
Also add an instructor-check for the course here. If I have the global instructor role (from another course) I could still access this on my own participation now.